pkg/vfile/vfile.go

```diff
@@ -214,3 +249,37 @@ func openHashedFile(path string, wantHash []byte, h hash.Hash) (*File, error) {
 	}
 	return f, nil
 }
+
+func CheckHashedContent(b *bytes.Reader, wantHash []byte, h hash.Hash) (*bytes.Reader, error) {
+	if len(wantHash) == 0 {
+		return b, ErrInvalidHash{
+			Err: ErrNoExpectedHash,
+		}
+	}
+
+	got, err := CalculateHash(b, h)
+	if err != nil {
+		return b, err
+	}
+
+	if !bytes.Equal(wantHash, got) {
+		return b, ErrInvalidHash{
+			Err: ErrHashMismatch{
+				Got:  got,
+				Want: wantHash,
+			},
+		}
+	}
+	return b, nil
+}
+
+func CalculateHash(b *bytes.Reader, h hash.Hash) ([]byte, error) {
+	// Hash the file.
+	if _, err := io.Copy(h, b); err != nil {
+		return nil, ErrInvalidHash{
```

Comment:
Isn't this more of a reader error than a hash error?

Reply:
The Copy error is preserved in the encapsulated error. The InvalidHash just defines a broader comparable type. I think it's okay for this case.

Reply:
Sounds good, thanks.